src/pudl/analysis/record_linkage/eia_ferc1_record_linkage_splink_model.py

```diff
@@ -0,0 +1,246 @@
+)
+
+
+@op(out={"eia_df": Out(), "ferc_df": Out()})
+def get_input_dfs(inputs):
+    """Get EIA and FERC inputs for the model."""
+    eia_df = (
+        inputs.get_plant_parts_eia_true()
+        .reset_index()
+        .rename(
+            columns={
+                "record_id_eia": "record_id",
+                "plant_name_eia": "plant_name",
+                "utility_name_eia": "utility_name",
+            }
+        )
+    )
+    ferc_df = (
+        inputs.get_plants_ferc1()
+        .reset_index()
+        .rename(
+            columns={
+                "record_id_ferc1": "record_id",
+                "plant_name_ferc1": "plant_name",
+                "utility_name_ferc1": "utility_name",
+            }
+        )
+    )
+    return eia_df, ferc_df
+
+
+@op
+def prepare_for_matching(df, transformed_df):
+    """Prepare the input dataframes for matching with splink."""
+    # replace old cols with transformed cols
+    for col in transformed_df.columns:
+        orig_col_name = col.split("__")[1]
+        df[orig_col_name] = transformed_df[col]
+    cols = ID_COL + MATCHING_COLS + EXTRA_COLS
+    df = df.loc[:, cols]
+    df["installation_year"] = pd.to_datetime(df["installation_year"], format="%Y")
+    df["construction_year"] = pd.to_datetime(df["construction_year"], format="%Y")
+    return df
+
+
+@op
+def get_training_data_df(inputs):
+    """Get the manually created training data."""
+    train_df = inputs.get_train_df().reset_index()
+    train_df = train_df[["record_id_ferc1", "record_id_eia"]].rename(
+        columns={"record_id_eia": "record_id_l", "record_id_ferc1": "record_id_r"}
+    )
+    train_df.loc[:, "source_dataset_r"] = "ferc_df"
+    train_df.loc[:, "source_dataset_l"] = "eia_df"
+    train_df.loc[
+        :, "clerical_match_score"
+    ] = 1  # this column shows that all these labels are positive labels
+    return train_df
+
+
+@op
+def get_model_predictions(eia_df, ferc_df, train_df):
+    """Train splink model and output predicted matches."""
+    settings_dict = {
+        "link_type": "link_only",
+        "unique_id_column_name": "record_id",
+        "additional_columns_to_retain": ["plant_id_pudl", "utility_id_pudl"],
+        "comparisons": COMPARISONS,
+        "blocking_rules_to_generate_predictions": BLOCKING_RULES,
+        "retain_matching_columns": True,
+        "retain_intermediate_calculation_columns": True,
+        "probability_two_random_records_match": 1 / len(eia_df),
+    }
+    linker = DuckDBLinker(
+        [eia_df, ferc_df],
+        input_table_aliases=["eia_df", "ferc_df"],
+        settings_dict=settings_dict,
+    )
+    linker.register_table(train_df, "training_labels", overwrite=True)
+    linker.estimate_u_using_random_sampling(max_pairs=1e7)
+    linker.estimate_m_from_pairwise_labels("training_labels")
+    # set a match probability threshold?
+    # do this at a bunch of different thresholds and print out stats
+    preds_df = linker.predict(threshold_match_probability=0.9)
```

Comment:
It would be nice if here we could get precision and recall stats for predictions at different match thresholds. Can we track this through experiment tracking infrastructure?